cips/cip-x.md

```diff
@@ -0,0 +1,77 @@
+| cip | XX (assigned by Editors) |
+| - | - |
+| title | Header Pruning for Light Nodes |
+| description | Mechanism to retain a fixed range of headers instead of whole history |
+| author | Hlib Kanunnikov ([@Wondertan](https://github.com/Wondertan)) |
+| discussions-to | <https://forum.celestia.org/t/cip-header-pruning-for-lns/1958> |
+| status | Draft |
+| type | Standards Track |
+| category | Data Availability |
+| created | 2025-03-21 |
+
+## Abstract
+
+Currently, every data availability (DA) node type synchronizes all historical headers starting from genesis (or other statically configured
+historical header) until the subjectively initialized head of the chain. We change that by adding a way to sync a
+constant size range of headers instead of the whole history.
+
+## Motivation
+
+Light nodes (LNs) store the whole history of ExtendedHeaders in their entirety, including ValidatorSet, Commit and data availability header (DAH). The motivation
```

Comment:
What is the motivation for LNs to keep any historical data in the first place?

Reply:
There is/was none. It's a techdebt that became a product issue and the time has come to clean it up.

Reply:
Right. Not a huge deal but I'd include that in the motivation along with the need to store at least the SampleWindow